DQMOffline/MuonDPG/src/BaseTnPEfficiencyTask.cc

```diff
@@ -187,7 +188,9 @@ bool BaseTnPEfficiencyTask::hasTrigger(std::vector<int>& trigIndices,
   float dR2match = 999.;
   for (int trigIdx : trigIndices) {
     const std::vector<std::string> trigModuleLabels = m_hltConfig.moduleLabels(trigIdx);
```

Comment:
@missirol I'm not sure how to give you a recipe to reproduce the problem, but if here I simply print out the module labels inside `trigModuleLabels`, these are the last filters in one of the event that I see in 12_4_0_pre3:

--> moduleLabels[296]: hltHpsSelectedPFTau27TightChargedIsolationAgainstMuonL1HLTMatched
--> moduleLabels[297]: hltHpsOverlapFilterIsoMu20TightChargedIsoPFTau27L1Seeded
--> moduleLabels[298]: hltBoolEnd
--> moduleLabels[299]: hltEcalRecHit@cpu
--> moduleLabels[300]: hltEcalUncalibRecHit@cpu
--> moduleLabels[301]: hltHbhereco@cpu
--> moduleLabels[302]: hltPixelTracksSoA@cpu
--> moduleLabels[303]: hltPixelVerticesSoA@cpu

while in 12_2_0_pre2 (that I used to test the first PR) I find e.g. :

--> moduleLabels[442]: hltPFTau27TrackLooseChargedIsoAgainstMuon
--> moduleLabels[443]: hltL1JetsHLTPFTauTrackLooseChargedIsolationAgainstMuonMatch
--> moduleLabels[444]: hltSelectedPFTau27LooseChargedIsolationAgainstMuonL1HLTMatched
--> moduleLabels[445]: hltOverlapFilterIsoMu20LooseChargedIsoPFTau27L1Seeded
--> moduleLabels[446]: hltBoolEnd

Reply:
> Since 12_4_0_pre3 it seems that additional filters appeared after hltBoolEnd [..].

@llunerti , okay, thanks. Now I understand. Those are not "additional filters" (or, `EDFilter`s), they are `EDProducer`s used in that trigger path. The reason they now appear at the end of this list goes back to a technical change in the HLT menu (those modules are now in `Task`s and not `Sequence`s, see https://github.com/cms-sw/cmssw/pull/37783 if interested in the details). I don't know this DQM plugin, but here it looks like you are trying to select the `EDFilter` of a given HLT Path before the final `hltBoolEnd`, and you could do that as follows (untested code):
```cpp
auto const& trigModuleLabels = m_hltConfig.moduleLabels(trigIdx);
uint trigModuleIndex = trigModuleLabels.size()-1;
for(uint midx = trigModuleIndex; midx == 0; --midx){
  auto const& moduleLabel = trigModuleLabels[midx];
  if (moduleLabel != "hltBoolEnd" and m_hltConfig.moduleEDMType(moduleLabel) == "EDFilter") {
    trigModuleIndex = midx;
    break;
  }
}
```

Reply:
@missirol thanks for the clarification! Shouldn't also work my solution? Or am I missing something?

Reply:
Sorry, I missed this message. My understanding is that your solution will work just as well for the cases you are considering (because the HLT paths monitored here will/should always have at least one filter, and the module before `hltBooEnd` would always be a `EDFilter`). At the same time, the implementation is a bit fragile, as these conditions are not true in general (for example, in trigger paths for Scouting, the module before `hltBoolEnd` is not a EDFilter). My opinion is that, with a slightly more solid implementation, you might save yourself (or others) trouble in the future (this PR itself wouldn't have been necessary if the implementation had been a bit more solid).